week1/week1-assigment/index.js

```diff
@@ -0,0 +1,52 @@
+// To run this file: node index.js
+// Then in your browser: http://localhost:8080
+var http = require('http');
+
+var port = 8080;
+let status = 10;
+var server = http.createServer();
+
+// Start the HTTP server, start listening for requests
+server.listen(port, function (error) {
+    if (error) {
+        console.log(error);
+    } else {
+        console.log('api listening on port', port);
+    }
+});
+
+// Create a event handler for "request"
+// this is an alternative way
+server.on('request', function (request, response) {
+
+
+    if (request.url === "/state") {
+        response.setHeader('content-type', 'text/html');
```

Comment:
you have `response.setHeader` and `response.end` the same in all the `if` cases, and `response.write` almost the same. Can you think of a way to refactor this to avoid so much repetition?

Reply:
function that generate this stuff is better right ?
less logic better you right

Reply:
Yes, a helper function would be a good idea (you can always look at the other PRs to see how the other students solved this problem)

Reply:
iwill check before i give up hahah iwill try to solve it my self first